lib/data/external/GCP/GcpManagedUpload.js

```diff
@@ -0,0 +1,432 @@
+const async = require('async');
+const stream = require('stream');
+const { lstatSync } = require('fs');
+const { errors } = require('arsenal');
+const { minimumAllowedPartSize, maximumAllowedPartCount } =
+    require('../../../../constants');
+const { createMpuList } = require('./GcpUtils');
+
+function sliceFn(size) {
+    this.array = [];
```

Comment:
Why use `this` and not a constant variable?

Reply:
Will fix this.